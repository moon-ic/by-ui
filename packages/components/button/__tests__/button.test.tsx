import { mount } from '@vue/test-utils'
import { describe, it, expect, vi } from 'vitest'
import Button from '../index.vue'

describe('Button Component', () => {

  it('should render correctly with default props', () => {
    const wrapper = mount(Button)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should apply the correct default type and size', () => {
    const wrapper = mount(Button)
    expect(wrapper.classes()).toContain('by-btn-primary')
    expect(wrapper.classes()).toContain('by-btn-small')
  })

  it('should apply the correct type when passed', () => {
    const wrapper = mount(Button, {
      props: {
        type: 'danger'
      }
    })
    expect(wrapper.classes()).toContain('by-btn-danger')
  })

  it('should apply the correct size when passed', () => {
    const wrapper = mount(Button, {
      props: {
        size: 'large'
      }
    })
    expect(wrapper.classes()).toContain('by-btn-large')
  })

  it('should be disabled when the "disabled" prop is true', () => {
    const wrapper = mount(Button, {
      props: {
        disabled: true
      }
    })
    expect(wrapper.classes()).toContain('disabled')
    expect(wrapper.attributes('disabled')).toBe('disabled')
  })

  it('should emit "click" event when clicked', async () => {
    const wrapper = mount(Button)
    const clickHandler = vi.fn()
    wrapper.vm.$on('click', clickHandler)

    await wrapper.trigger('click')
    expect(clickHandler).toHaveBeenCalled()
  })
})
