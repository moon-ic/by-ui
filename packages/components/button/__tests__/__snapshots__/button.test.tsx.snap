// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > should render correctly with default props 1`] = `
<button
  class="by-btn by-btn-primary by-btn-small"
  data-v-76618e17=""
>
  
  
</button>
`;
